### PR TITLE
Levelbuilder Clean Up: Code Area Editors

### PR DESCRIPTION
**Pull request by dmcavoy**
Similar to https://github.com/code-dot-org/code-dot-org/pull/35035 this groups together all the toolbox and code area editors together.

### Droplet style level

code-area

### Blockly Style level

code-area-blockly

### CODE REVIEW

**Comment by davidsbailey** [comment]
this doesn't look right -- the PR is full of commits from other parts of the staging branch. perhaps you just need to merge latest staging into this branch to fix it?

**Comment by dmcavoy** [comment]
> this doesn't look right -- the PR is full of commits from other parts of the staging branch. perhaps you just need to merge latest staging into this branch to fix it?

Sorry about that. I think I fixed it now

File: dashboard/app/views/levels/editors/_droplet.html.haml [hunk 1]
```diff
@@ -1,23 +1,36 @@
...
   = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :droplet_tooltips_disabled, description: "Hide droplet documentation tooltips"}
-.field
   = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :lock_zero_param_functions, description: "Don't allow parameters to be added to user-defined functions with no parameters"}
-.field
+
+  - if @level.is_a? Applab
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :makerlab_enabled, description: "Enable Maker APIs"}
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :autocomplete_palette_apis_only, description: "Autocomplete palette APIs only"}
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :execute_palette_apis_only, description: "Execute palette APIs only"}
+    = render partial: 'levels/editors/fields/checkboxes', locals: {f: f, field_name: :beginner_mode, description: "Beginner mode for loops"}
```
Comment (davidsbailey) [inline dashboard/app/views/levels/editors/_droplet.html.haml:12]: it's a little weird to see fields pulled out of an applab-specific partial, only to be put behind an applab conditional. but it's more important to arrange the fields the way that the curriculum team wants than it is to keep conditionals out of our code, so this still seems like an improvement to me.

**Review by davidsbailey — APPROVED** [review]

File: dashboard/app/views/levels/editors/_droplet.html.haml (hunk 1 above)
Comment (dmcavoy) [reply dashboard/app/views/levels/editors/_droplet.html.haml:12]: I agree. I'm kinda trying to organize in small chunks that make sense and I think there will be some moving things to possibly move them again in order to get this right. The mistake I made last time was trying to make it right on the first go and that was too hard to follow for reviews.

**Review by uponthesun — APPROVED** [review]

File: dashboard/app/views/levels/editors/_droplet.html.haml (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/views/levels/editors/_droplet.html.haml:12]: No worries, makes sense! FWIW the most vocal feedback I recall receiving last time was from Elizabeth objecting to having things being hidden by default, since that broke her flow of doing a text search to find the field that she was looking for. I think you're doing a fine job of avoiding that here by making all the expand/collapse controls be expanded by default.
